src/main/java/com/google/api/codegen/config/FieldConfigGenerator.java

```diff
@@ -62,8 +62,24 @@
       if (parameterList.size() <= FLATTENING_THRESHOLD) {
         result.put(CONFIG_KEY_FLATTENING, createFlatteningConfig(parameterList));
       }
-      result.put(CONFIG_KEY_REQUIRED_FIELDS, new LinkedList<String>(parameterList));
     }
+
+    // For the required fields, apart from ignoring pagination parameters, ignore any boolean
+    // fields. Those need always be optional, for requiring one to have a value other than the
+    // default (false) is requiring it to be set to true, which doesn't make sense.
+    List<String> requiredParameters = new LinkedList<>();
+    for (Field field : message.getFields()) {
+      String fieldName = field.getSimpleName();
+      if (ignoredFields.contains(fieldName)) {
+        continue;
+      }
+      if (field.getType().getKind() == Type.TYPE_BOOL && !field.getType().isRepeated()) {
+        continue;
+      }
```

Comment:
Also drop it if `field.isDeprecated()`. Although that's tricky because deprecating a field is a non-breaking change that could break our surface if we act on it.